### DISCUSSION TITLE
IKEA Styrbar E2001/E2002, doesn't work properly after upgrading to 2.4.11

### DISCUSSION THREAD

**Discussion by misternic** [Koenkk/zigbee2mqtt#23243] [Devices]
After updating our Styrbar remote control to version 2.4.11 (Firmware build date 20240408), they started strange behaviour. When briefly pressing the brightness up or brightness down buttons, they no longer trigger the "on" status. Instead, they alternately send "brightness_move_up" and "brightness_stop" commands, even though the buttons are not being held down. The same issue occurs with other buttons; for example, the "arrow_left_hold" and "arrow_left_release" states are triggered, but not the "arrow_left_click".

Additionally, we noticed that if the last status was a "hold" state, no other button presses are recognized until the same button is pressed again to trigger the "release" state.

Here are the steps I have already taken to resolve the issue:

* Forcefully removed the remote controls from the system
* Restarted Zigbee2Mqtt
* Removed and reinserted the batteries
* Rejoined the remotes to the network

These measures temporarily resolve the issue for a few hours, but the remote controls revert to the described behavior.

We currently have seven Styrbar remote controls in use, and it would be unfortunate if we had to return them to Ikea.

I am unsure how to proceed. I have considered downgrading to the previous version, but I have not found any information on how to do this. 

I would greatly appreciate any help or suggestions.

**Zigbee2MQTT version:** 1.38.0
**Coordinator:** Home Assistant SkyConnect, EZSP 7.4.2.0 build 0

**Comment by EricV-hue** [comment]
I have the same problem unfortunately. I have two remotes, both on firmware 2.4.11.
The behaviour is like described in the first post. I also tried the same actions to resolve. No succes.

any thoughts? Can i reverse to a previous firmware, somehow?

Zigbee2MQTT: 1.39.0-1
HomeAssisstant: latest, Odroid N2+
Coordinator: Slaesh CC2652R

**Comment by Koenkk** [comment]
Does reconfiguring the device via the z2m frontend fix the issue? Press the yellow refresh icon and make sure to wakeup the device right before doing so

**Comment by EricV-hue** [comment]
I just rejoined the remotes. I now use an automation to switch a light. Previous setup was with a direct binding to the light.
When the problem reoccurs i will try your suggestion and see what happens

**Comment by misternic** [comment]
Reconfiguring does not resolve the issue. Here are the logs and binding:

```
Info 2024-07-07 13:25:53MQTT publish: topic 'zigbee2mqtt/Remote Control Bett/action', payload 'brightness_move_up'
Info 2024-07-07 13:25:54MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"brightness_stop","battery":76,"linkquality":160,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:25:54MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"","battery":76,"linkquality":160,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:25:54MQTT publish: topic 'zigbee2mqtt/Remote Control Bett/action', payload 'brightness_stop'
Info 2024-07-07 13:25:56MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"brightness_move_up","action_rate":83,"battery":76,"linkquality":160,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:25:56MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"","battery":76,"linkquality":160,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:25:56MQTT publish: topic 'zigbee2mqtt/Remote Control Bett/action', payload 'brightness_move_up'
Info 2024-07-07 13:25:57MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"brightness_stop","battery":76,"linkquality":156,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:25:57MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"","battery":76,"linkquality":156,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:25:57MQTT publish: topic 'zigbee2mqtt/Remote Control Bett/action', payload 'brightness_stop'
Info 2024-07-07 13:25:57Configuring 'Remote Control Bett'
Info 2024-07-07 13:26:02Successfully configured 'Remote Control Bett'
Info 2024-07-07 13:26:03MQTT publish: topic 'zigbee2mqtt/bridge/response/device/configure', payload '{"data":{"id":"Remote Control Bett"},"status":"ok","transaction":"u8dca-3"}'
Info 2024-07-07 13:26:04MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"brightness_move_up","action_rate":83,"battery":75,"linkquality":168,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:26:04MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"","battery":75,"linkquality":168,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:26:04MQTT publish: topic 'zigbee2mqtt/Remote Control Bett/action', payload 'brightness_move_up'
Info 2024-07-07 13:26:44MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"brightness_stop","battery":75,"linkquality":164,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:26:44MQTT publish: topic 'zigbee2mqtt/Remote Control Bett', payload '{"action":"","battery":75,"linkquality":164,"update":{"installed_version":33816593,"latest_version":33816593,"state":"idle"},"update_available":null}'
Info 2024-07-07 13:26:44MQTT publish: topic 'zigbee2mqtt/Remote Control Bett/action', payload 'brightness_stop'
```

![image](https://github.com/Koenkk/zigbee2mqtt/assets/50358374/048051c2-b10f-4b7d-b124-cbc0e0eaeab8)

**Comment by EricV-hue** [comment]
Problem is back. Reconfiguring reported to be ”succesfull”. But there was no change. So reconfiguring does not help.

The actions for the buttons is as descibed in the opening post.
When i do a short press of the brightness up button the action is ‘brightness_move_up’
A second short press gives ‘brightness_stop’
A long press of the same button results in a sequence: ‘brightness_move_up’ -> ‘brightness_stop‘

I did not use this remote for about a day and then got this change of behaviour.

**Comment by EricV-hue** [comment]
I started an experiment. I have an older model Trädfri hub. I have bound a remote and a lightbulb.
Up to now (about 16 hours later) the remote works as should be.

**Comment by misternic** [comment]
As I have configured the buttons for different functions in Home Assistant, such as playing radio through a speaker, pairing with the hub doesn't work for me. After researching, I found the following information to downgrade the remotes:

- Zigbee2Mqtt retrieves information from IKEA products at http://fw.ota.homesmart.ikea.net/feed/version_info.json, which only shows current firmware versions.

- Using the Wayback Machine, we found the old version at https://web.archive.org/web/20220620220011/http://fw.ota.homesmart.ikea.net/feed/version_info.json and the corresponding OTA file:

  - http://fw.ota.homesmart.ikea.net/global/GW1.0/01.19.026/bin/10078031-zingo_kt_styrbar_remote-1.0.024.ota.ota.signed for version 1.0.024, which is still downloadable.

- We then created a local index file for the firmware files (see https://www.zigbee2mqtt.io/guide/usage/ota_updates.html#local-ota-index-and-firmware-files).

    - Clone the repository https://github.com/Koenkk/zigbee-OTA and run the following command with Node.js:

    - `node scripts/add.js http://fw.ota.homesmart.ikea.net/global/GW1.0/01.19.026/bin/10078031-zingo_kt_styrbar_remote-1.0.024.ota.ota.signed`

    - This creates the `index.json` file, which should be placed in the same folder as the `configuration.yaml` of Zigbee2Mqtt.

    - The end of the index.json should look like this:

```diff
...
	},
    {
        "fileVersion": 65572,
        "fileSize": 194002,
        "manufacturerCode": 4476,
        "imageType": 4555,
+       "force": true,
        "sha512": "732ed02c39087dfffbe1d6e7adaf23851ff8dc695fe1e0c85daa5b7e1a3bea85796181d6da02b695685bc35470cf0c69cb06cd73c4bb51c67f89a198b1dac78f",
        "url": "http://fw.ota.homesmart.ikea.net/global/GW1.0/01.19.026/bin/10078031-zingo_kt_styrbar_remote-1.0.024.ota.ota.signed"
	}
}
```

- Add or change the following to `configuration.yaml`:

```yaml
ota:
  zigbee_ota_override_index_location: index.json
  disable_automatic_update_check: true
```

- Add `"force": true` to the ota object for styrbar in `index.json`.
  
- (optional) Ensure the `fileVersion` 65572 in `index.json` matches the OTA file (determine with `./zigpy ota info 10078031-zingo_kt_styrbar_remote-1.0.024.ota.ota.signed` after downloading the file with `wget http://fw.ota.homesmart.ikea.net/global/GW1.0/01.21.031/bin/styrbar_dimmer_zingo-0x11cb-2.4.11-prod.ota.ota.sig`, see https://github.com/zigpy/zigpy, `pip install zigpy-cli`).


- Since IKEA devices do not use the generic Zigbee OTA configuration, modify `ikea.js` and add `ota: ota.zigbeeOTA` in the object for E2001/E2002 and remove the line with `(0, ikea_1.ikeaOta)()` (see https://github.com/Koenkk/zigbee2mqtt/issues/18023#issuecomment-1624316644). This steps depend on how Zigbee2Mqtt is run; in my case, it is a Docker container:

  - Access the Zigbee2Mqtt Docker container (`sudo docker exec -it <container_id> /bin/sh`).

  - Install Nano with `apk add nano`.

  - Edit the file via `nano /app/node_modules/zigbee-herdsman-converters/devices/ikea.js`:

```diff
 {
        zigbeeModel: ['Remote Control N2'],
        model: 'E2001/E2002',
        vendor: 'IKEA',
        description: 'STYRBAR remote control',
+       ota: ota.zigbeeOTA,
        extend: [
            (0, ikea_1.ikeaConfigureStyrbar)(),
            (0, modernExtend_1.identify)({ isSleepy: true }),
            (0, ikea_1.styrbarCommandOn)(),
            (0, modernExtend_1.commandsOnOff)({ commands: ['off'], bind: false }),
            (0, modernExtend_1.commandsLevelCtrl)({ commands: ['brightness_move_up', 'brightness_move_down', 'brightness_stop'], bind: false }),
            (0, ikea_1.ikeaArrowClick)({ styrbar: true, bind: false }),
            (0, ikea_1.ikeaBattery)(),
-           (0, ikea_1.ikeaOta)(),
        ],
    },
```

- Restart the entire Docker container; just restarting Zigbee2Mqtt is not sufficient.

We have tested this on three remotes, and it has been working since yesterday.

We have additional Styrbar remotes and can reinstall the latest firmware version if needed to further debug the issue if someone with more expertise is available.

Hope this information is useful for anyone.

**Comment by MacTheFork** [comment]
I have 2 styrbar remotes, both were on firmware 2.4.11.  Recently I noticed that one wasn't behaving according to the blueprint automation I was using.  On further investigation it shows the same behaviour as described by the OP. 

I see from the [release notes](https://ww8.ikea.com/ikeahomesmart/releasenotes/releasenotes.html) that as of 8th August there is a firmware 2.4.16 available.  I updated the device that wasn't working and it now behaves as expected. 

Should be worth trying to update firmware to this version rather than revert to a previous version.

**Comment by csepiy** [comment]
I have z2m 1.40.1 and unfortunatelly upgraded my styrbar bound to a group of bulbs from version 1.x to 2.4.16. Now styrbar is not working at all. I removed it from z2m and repaired (it wasn't easy). Last seen hasn't updated so far, even if I press the buttons or remove/reinsert batteries. Battery and action fileds are N/A. Tried to bind to bulbs separately, but got errors all the time.

**Comment by schitzosteve2** [comment]
I have latest z2m running with styrbar 2.4.16. It stopped working around 28.11. for unknown reason. I can force-remove the device and join it again to z2m, but after initial connection, it's not reacting. Tried several times, replaced batteries. No blinking when buttons are pressed, no events on z2m. Any tips what I can share to debug?

**Reply by ChristophTrommler** [reply]
I made everything brandnew and after that facing the exact same issue, did you manage to solve the problem?

Edit: Removed batteries for 1h, put fully charged in, then suddenly non-force remove worked when pushing every button over and over, stopped z2m, reset styrbar, started z2m, pushed 10x fast again and then keep pushing slow and front buttons until I got the device with battery state. Took me some time of trying and pushing buttons but if someone facing the same problem, keep trying:)

**Comment by janmalec** [comment]
In case anyone has the same issue, mine was quite trivial. Mine was not working when updating to 2.4.16 even after multiple repairs. When examining logs I noticed 
`[homeassistant.components.recorder.entity_registry] Cannot migrate history for entity_id update.ikea_of_sweden_remote_control_n2_firmware to update.entrance_remote_firmware because the new entity_id is already in use`.
I tried repairing and not remaining it and it worked. Not sure if this was the only cause of trouble or not.

**Reply by prehor** [reply]
> `Cannot migrate history for entity_id update.ikea_of_sweden_remote_control_n2_firmware to update.entrance_remote_firmware because the new entity_id is already in use`.

This issue occurs if you rename entity_id. The correct procedure is described here [home-assistant/core/issues/115747#2560857963](https://github.com/home-assistant/core/issues/115747#issuecomment-2571530128)

**Comment by olylak** [comment]
I have upgraded to version 2.4.16. Everything looks good - all actions from the device are reflected in z2m. 
But if I bind it to a group it does nothing, command are sent but destination devices are not triggered.
No error in the log. 
Before version upgrade everything was working fine.

Same control but with lower version 1.0.024 works fine with the same group.
